client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/OperationsFactory.java

```diff
@@ -56,6 +58,8 @@ public class OperationsFactory implements HotRodConstants {
 
    private Codec codec;
 
+   private Codec safePingVersionCodec;
```

Comment:
I think we can rename these two, something in the lines of configuredCodec and negotiatedCodec. Only now I realized that this class was supposed to be immutable, but I am not sure if we need to add something for visibility on the negotiated and make the codec final.

Reply:
The codec itself was not immutable before this contribution, the idea was to update the codec when the negotiated version. I think